plan/aggregation_push_down.go

```diff
@@ -281,13 +281,13 @@ func (a *aggPushDownSolver) pushAggCrossUnion(agg *Aggregation, unionSchema expr
 		newAggFunc := aggFunc.Clone()
 		newArgs := make([]expression.Expression, 0, len(newAggFunc.GetArgs()))
 		for _, arg := range newAggFunc.GetArgs() {
-			newArgs = append(newArgs, expression.ColumnSubstitute(arg, unionSchema, expression.Column2Exprs(unionChild.GetSchema().Columns)))
+			newArgs = append(newArgs, expression.ColumnSubstitute(arg, unionSchema.Columns, expression.Column2Exprs(unionChild.GetSchema().Columns)))
 		}
 		newAggFunc.SetArgs(newArgs)
 		newAgg.AggFuncs = append(newAgg.AggFuncs, newAggFunc)
 	}
 	for _, gbyExpr := range agg.GroupByItems {
-		newExpr := expression.ColumnSubstitute(gbyExpr, unionSchema, expression.Column2Exprs(unionChild.GetSchema().Columns))
+		newExpr := expression.ColumnSubstitute(gbyExpr, unionSchema.Columns, expression.Column2Exprs(unionChild.GetSchema().Columns))
```

Comment:
After adding unique key info into schema, columns2expr will be more accurate than schema2expr.